### PR TITLE
Stream in CAFS data for read only FS mounted bundles

### PR DESCRIPTION
**Pull request by kerneltime**
Signed-off-by: Ritesh H Shukla <ritesh@oneconcern.com>

### CODE REVIEW

File: pkg/storage/localfs/store.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func (rc readCloser) Close() error {
 }
 
 func (l *localFS) Put(ctx context.Context, key string, source io.Reader, exclusive bool) error {
+	// TODO: Change this implementation to use rename to put file into place.
```
Comment (ransomw1c) [inline pkg/storage/localfs/store.go]: see #162 for a decorator pattern approach

**Comment by ransomw1c** [comment]
overall it looks quite good.  adding a `ReadAt` to `Bundle` was not an approach i considered.

File: pkg/cafs/reader.go [hunk 2]
```diff
@@ -159,7 +161,46 @@ func (r *chunkReader) WriteTo(writer io.Writer) (n int64, err error) {
...
+		// Read first leaf
+
+		read, err = rdr.ReadAt(p[n:], offset)
+		n += read
+		index++
+		offset = 0
+		if err != nil && !strings.Contains(err.Error(), "EOF") {
+			return
+		}
+		if (len(p) == n) || (index >= int64(len(r.keys))) {
```
Comment (ransomw1c) [inline pkg/cafs/reader.go]: what is the intent of `len(p) == n`?  when is this false?

**Comment by ransomw1c** [comment]
global style comment: the "naked return" wherein return variables are specified in the function declaration is suggested to be used in terse functions only according to the Go Tour.

File: pkg/storage/localfs/store.go (hunk 1 above)
Comment (kerneltime) [reply pkg/storage/localfs/store.go]: Can you file an issue under harmonization? Don't want to miss this.

File: pkg/cafs/reader.go (hunk 2 above)
Comment (kerneltime) [reply pkg/cafs/reader.go]: len(p) can be greater than n till reads are complete.

**Comment by kerneltime** [comment]
Moving this to https://github.com/oneconcern/datamon/pull/182

**Pull request closed without merging**